Winleafs-Server/Winleafs.Server.Services/SpotifyService.cs

```diff
@@ -0,0 +1,218 @@
+        }
+
+        public async Task<Dictionary<string, string>> GetPlaylists(string applicationId)
+        {
+            var webAPI = await GetRefreshedSpotifyWebAPI(applicationId);
+            var profile = await webAPI.GetPrivateProfileAsync();
+
+            CheckResponse(profile);
+
+            var info = await webAPI.GetUserPlaylistsAsync(profile.Id);
+
+            return info.Items.ToDictionary(playlist => playlist.Id, playlist => playlist.Name);
+        }
+
+        public async Task Disconnect(string applicationId)
+        {
+            var user = await _userService.FindUserByApplicationId(applicationId);
+
+            if (user != null)
+            {
+                user.SpotifyAccessToken = null;
+                user.SpotifyExpiresOn = null;
+                user.SpotifyRefreshToken = null;
+
+                _userService.Update(user);
+
+                await _context.SaveChangesAsync();
+            }
+        }
+
+        public async Task<bool> IsConnected(string applicationId)
+        {
+            try
+            {
+                var webAPI = await GetRefreshedSpotifyWebAPI(applicationId);
+                var playbackContext = await webAPI.GetPlaybackAsync();
+
+                CheckResponse(playbackContext);
+
+                return playbackContext.Error == null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        #region Privates
+        private string GetSpotifyAuthorizationHeader()
+        {
+            var spotifyauthorizationBytes = Encoding.UTF8.GetBytes($"{SpotifyClientInfo.ClientID}:{SpotifyClientInfo.ClientSecret}");
+            return Convert.ToBase64String(spotifyauthorizationBytes);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="SpotifyWebAPI"/> object for the given
+        /// <paramref name="applicationId"/> to query the Spotify API.
+        /// Refreshes the user's access token if it expired.
+        /// </summary>
+        private async Task<SpotifyWebAPI> GetRefreshedSpotifyWebAPI(string applicationId)
+        {
+            var user = await _userService.FindUserByApplicationId(applicationId);
+
+            if (user == null)
+            {
+                throw new InvalidApplicationIdException("No user was found for the given application id");
+            }
+
+            if (user.SpotifyAccessToken == null)
+            {
+                throw new SpotifyNotConnectedException("The user is not connected to Spoify; no Spotify information is saved for the user.");
+            }
+
+            if (DateTime.Now > user.SpotifyExpiresOn)
+            {
+                await RefreshToken(user);
+            }
+
+            return new SpotifyWebAPI
+            {
+                TokenType = "Bearer", //TokenType is always bearar, hence we do not need to save it on a user object
```

Comment:
Bearar >  Bearer

Reply:
Also try to put comments above code, it's in my opinion a better standard and easier to read